pkg/proxy/endpoints.go

```diff
@@ -197,14 +200,14 @@ type UpdateEndpointMapResult struct {
 	StaleServiceNames []ServicePortName
 	// List of the trigger times for all endpoints objects that changed. It's used to export the
 	// network programming latency.
-	LastChangeTriggerTimes []time.Time
+	LastChangeTriggerTimes map[types.NamespacedName][]time.Time
```

Comment:
I don't understand why this is needed?
[or at the very least, it doesn't seem to be related to handling deletedEndpoints, so please remove it from this PR]

Reply:
This change is needed to improve logging. So far we've only logged latency, but not the name of the endpoint, making the problem hard to debug. 

I'm fine with moving this to a separate PR - please let me know.

Reply:
Yes - please move it to a separate PR - please don't mix two separate things in a single PR.